src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -33,10 +40,43 @@
   /** Create a new Businesses instance
   * @param businesses businesses from SmallCityService
   **/
+  
   public BusinessesService(List<Listing> businesses) {
     this.businesses = businesses;
   }
 
+  public PreparedQuery getBigBusinessFromDatabase(){
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("BigBusinesses");
+    PreparedQuery dateabaseResults = datastore.prepare(query);
+    return dateabaseResults;
+  }
+
+  public List<Listing> removeBigBusinessesFromResults(PreparedQuery dateabaseResults){
+    Iterator<Listing> businessesList =  businesses.iterator();
+    Entity entity;
+    Boolean untilBigBusinessIsFound = true;
```

Comment:
The "until" here doesn't add much value. The convention is to prefix boolean variable names with the 3rd person of be or have, so "is" or "has". This can easily become "isFound" by that convention, or bigBusinessIsFound if the desire is to have more context, and be more specific.

Reply:
I changed the variable name, but the only reason I left the "until " part as the variable is equal to true initially. This shows that when the big business is found it should change to false, or at least in my head is how I am thinking about this.